zap/src/irgen/mod.rs

```diff
@@ -469,6 +575,7 @@ impl Display for Expr {
 
 			Self::Var(var) => write!(f, "{var}"),
 			Self::Num(num) => write!(f, "{num}"),
+			Self::BinaryNum(num) => write!(f, "0b{num:016b}"),
```

Comment:
I feel like we should instead of padding to a `u16`, we should pad to the nearest type (u8, u16, u32)

As in, this should really be padded to a `u8`:
```lua
			value[1] = bit32.btest(bool_5, 0b0000000000000001)
			value[2] = bit32.btest(bool_5, 0b0000000000000010)
			value[3] = bit32.btest(bool_5, 0b0000000000000100)
			value[4] = bit32.btest(bool_5, 0b0000000000001000)
			value[5] = bit32.btest(bool_5, 0b0000000000010000)
			value[6] = bit32.btest(bool_5, 0b0000000000100000)
			value[7] = bit32.btest(bool_5, 0b0000000001000000)
			value[8] = bit32.btest(bool_5, 0b0000000010000000)
```